### ISSUE TITLE
FB not working behind a reverse proxy over https in Docker

### ISSUE THREAD

**Issue by tkurki** [SignalK/freeboard-sk#159]
I have SK running in Docker behind nginx that handles https. Freeboard is not working, both http and ws are rejected:

![image](https://github.com/SignalK/freeboard-sk/assets/1049678/60b29d86-a45f-4e18-9392-64b9cdfe531b)

![image](https://github.com/SignalK/freeboard-sk/assets/1049678/599b397e-069f-4e76-8567-116463573ee2)

**Comment by panaaj** [comment]
Have you checked "behind proxy" in the settings?
![Screenshot_20240610-094717.png](https://github.com/SignalK/freeboard-sk/assets/38519157/4401cb49-5a17-47c2-a2e7-d043433bc327)

**Comment by tkurki** [comment]
Gmail ate your reply, but that worked.

What does it actyally do? I would rather have it work out of the box.

**Comment by panaaj** [comment]
By default Freeboard-Sk uses the endpoints returned in the response to `/signalk` 
```
  "endpoints": {
    "v1": {
      "version": "2.8.2",
      "signalk-http": "http://192.168.86.32:3000/signalk/v1/api/",
      "signalk-ws": "ws://192.168.86.32:3000/signalk/v1/stream",
      "signalk-tcp": "tcp://192.168.86.32:8375"
    }
  },
  "server": {
    "id": "signalk-server-node",
    "version": "2.8.2"
  }
}
```
Checking the box uses the protocol:host:port from the browser location rather than the response.

**Comment by tkurki** [comment]
Ah, own footgun. Hmm, need to think of a way to solve this.

**Comment by tkurki** [comment]
What if FB primarily tried the server and protocol it was loaded from and secondarily tried the discover endpoint? I think discovery is more useful for discovered servers and standalone clients, less so for webapps.

**Comment by panaaj** [comment]
Sounds fair.
Also sounds like a great reason to create a Signal K client apps best practice guide in the server docs

**Comment by panaaj** [comment]
Will set proxy mode as the default.

**Comment by panaaj** [comment]
closed by v2.11.2
